patterns/vsi/module/main.tf

```diff
@@ -34,6 +34,7 @@ module "landing_zone" {
   teleport_config_data                   = local.env.teleport_config
   teleport_vsi                           = local.env.teleport_vsi
   vpc_placement_groups                   = local.env.vpc_placement_groups
+  user_data                              = var.user_data
```

Comment:
As stated in the root level variables file, I don't think this should be its own separate input variable, as this property already exists within the VSI input map.

Instead, your new input variable for the pattern should be used to set the existing `user_data` property for the individual VSI entries in that map, somewhere in the config.tf: https://github.com/terraform-ibm-modules/terraform-ibm-landing-zone/blob/main/patterns/vsi/module/config.tf#L85